components/kyma-environment-broker/internal/storage/driver/postsql/operation.go

```diff
@@ -40,6 +40,39 @@ func (s *operations) InsertProvisioningOperation(operation internal.Provisioning
 	return s.insert(dto)
 }
 
+// InsertOperation insert new Operation to storage
+func (s *operations) InsertOperation(operation internal.Operation) error {
+	dto, err := s.operationToDTO(&operation)
+
+	if err != nil {
+		return errors.Wrapf(err, "while inserting operation (id: %s)", operation.ID)
+	}
+
+	return s.insert(dto)
+}
+
+// GetOperationByInstanceID fetches the latest Operation by given instanceID, returns error if not found
+func (s *operations) GetOperationByInstanceID(instanceID string) (*internal.Operation, error) {
+
+	op, err := s.getByInstanceID(instanceID)
+	if err != nil {
+		return nil, err
+	}
+
+	var operation internal.Operation
+	err = json.Unmarshal([]byte(op.Data), &operation)
+	if err != nil {
+		return nil, errors.New("unable to unmarshall provisioning data")
+	}
+
+	ret, err := s.toOperation(op, operation.InstanceDetails)
+	if err != nil {
+		return nil, errors.Wrapf(err, "while converting DTO to Operation")
+	}
+
```

Comment:
unarshalling (lines 63-66) must be done after lines 68-71 because toOperation creates new instance of Operation and all fields from marshalled json which are not in InstanceDetails will be lost

Reply:
my idea is: 
```
var operation internal.Operation


	ret, err := s.toOperation(op, operation.InstanceDetails)
	if err != nil {
		return nil, errors.Wrapf(err, "while converting DTO to Operation")
	}

var operation internal.Operation
	err = json.Unmarshal([]byte(op.Data), &operation)
	if err != nil {
		return nil, errors.New("unable to unmarshall provisioning data")
	}

	ret, err := s.toOperation(op, operation.InstanceDetails)
	if err != nil {
		return nil, errors.Wrapf(err, "while converting DTO to Operation")
	}
	```

Reply:
Will be corrected by @piotrmiskiewicz in PR-1870.